### PR TITLE
ISSUE-89: Image Upload Bugs

### LINKED ISSUE #89: Error uploading images to canvas bg
When uploading images I get an error in the browser console
image

To reproduce:
- I uploaded an svg (which usually works)
- then a png and a jpg (which both throw the error)

The image sometimes flickers on the screen before disappearing

### PR DESCRIPTION
**Pull request by Ammar-T**
## Details

- Added a new `Remove Background Image` button which is self-explanatory
- Restricted image uploads to `.png` and `.svg` since `.jpeg/.jpg` are very large when encoded to strings

Note: In the future, we should look to move to alternate image storing service which will allow us to store larger images. Or, find a way to compress `.jpg` images.

Closes #89

### CODE REVIEW

File: src/app/utils/FabricUtils.ts [hunk 1]
```diff
@@ -157,4 +157,24 @@ export class FabricUtils {
...
+          height = Math.abs(vptCoords.br.y - vptCoords.tr.y)
+        }
+
+        return {
+          top: vptCoords?.tl.y,
+          left: vptCoords?.tl.x,
+          width: width,
+          height: height,
+          scaleX: width / (img.width ?? 0),
+          scaleY: height / (img.height ?? 0)
```
Comment (StevenWuzz) [inline src/app/utils/FabricUtils.ts:177]: Is the division by zero (when img.width or img.height is null or undefined) intentional here?

**Comment by StevenWuzz** [comment]
> find a way to compress .jpg images #

Just wanted to mention here since you guys have probably taken CSCC73: The Huffman Algorithm could be used to compress such an image (and it is actually the best image compression algorithm AFAIK)

**Comment by Ammar-T** [comment]
@StevenWuzz Haven't taken C73 yet unfortunately, but thanks for pointing that out. I'm gonna look forward to it!

File: src/app/utils/FabricUtils.ts (hunk 1 above)
Comment (Ammar-T) [reply src/app/utils/FabricUtils.ts:177]: Yeah, the `0` should be a `1` here, good catch.

**Review by StevenWuzz — APPROVED** [review]

**Review by JoelWiebe — APPROVED** [review]
Thanks @Ammar-T, looks great

Tested:
- Adding background image
- Replacing background image
- Removing background image
- jpg cannot be uploaded

**Review by l3n0ire — APPROVED** [review]

**Pull request merged**